Feature RLS001 restaurant list

andress20 opened the pull request:
Create Restaurnt List Page.

The feature is just to show the current resturants in the platform. List them without any order (for now). The list getting fill with mock data in a separete file. 

Restaurant item only include Logo, Name and access button (no functional just the button). And the list is located leaving space in the left side of the screen for a future menu.

Finally we are link it the list with the button in the header component.

- [x] Restaurants.js (page in the router component)
- [x] Restaurants.js ( function whom render the list)
- [x] Data.js (mock data)
- [x] Header.js (to link the list in the header menu)

![ResturantList](https://user-images.githubusercontent.com/8940946/99000431-42c2e600-2507-11eb-93b6-ff10670c7744.png)

andress20 committed:
Create Restaurant component

andress20 committed:
Restaurant names

andress20 committed:
create list restaurants

andress20 committed:
before merge feature-REG001

andress20 committed:
merge Sig-Up component to my local branch

andress20 committed:
create restaurant list and link it in header

simonhoyos commented on alamesa/src/components/Restaurant.js:
```diff
@@ -0,0 +1,49 @@
+const Img = styled.img`
+	float: left;
+	width:  120px;
+  height: 120px;
```
ojo con la identación. los demás archivos están identados a 2 espacios y los nuevos archivos estan identados a 4 espacios o un tab

andress20 replied:
Si tengo super presente la identación, de hecho la vulevo a revisar y esta bien en mi editor. La tengo a 2 espacios (el tab tambien marca 2 espacios). Arreglaré los otros comentarios y a la proxima espero salga bien la identación. Gracias!

![identacion](https://user-images.githubusercontent.com/8940946/99011458-2b412880-251a-11eb-9738-9e11addfea38.png)

simonhoyos commented on alamesa/src/components/Restaurant.js:
```diff
@@ -0,0 +1,49 @@
+	margin: 0px 10px;
+`
+const Button = styled.button`
+	type: button;
```
el tipo del botón es un atributo o prop del componente y no una propiedad de css.

simonhoyos commented on alamesa/src/components/Restaurant.js:
```diff
@@ -0,0 +1,49 @@
+			/>
+			<RestaurantName>
+				<h1>{name}</h1>
+				<Button>
```
Acá es donde tienes que poner el tipo

simonhoyos commented on alamesa/src/components/Restaurant.js:
```diff
@@ -0,0 +1,49 @@
+	float: left;
+	width:  120px;
+  height: 120px;
+  object-fit: cover;
```
Object-fit no tiene soporte en el 100% de los navegadores y en algunas versiones. Te recomiendo probarlo en Safari más que todo

andress20 replied:
La otra opción que encontré para cambiar `Object-fit` es `background-size`. Aunque probando veo que si la elimino del todo no me afecta en la imagen que estoy colocando. La coloqué para que independientemente del tamaño de imagen que pasaramos ahi, esta se acomodara al espacio que tenemos designado para el logo del restaurante. Me recomiendas eliminarla o dejar el background?

No encontré forma de instalar safari en Linux (Ubuntu) pero si pude instalar epiphany que maneja el mismo motor de renderizado de safari **WebKit** a lo mejor no es 100% fidedigno pero me ayudará a tener una idea del comportamiento en safari por ahora.

simonhoyos replied:
super, no hay problema. Por ahora solo es para que lo tengan en cuenta, que a veces hay que probar en otros navegadores para ver que todo funcione

simonhoyos commented on alamesa/src/pages/Restaurants.js:
```diff
@@ -0,0 +1,49 @@
+					<Section>
+						{!!name && name.length > 0 && name.map(({name}) => {
+							return (
+								<Restaurant 
```
Recuerda pasarle el prop `key` a las listas. En el mock data podrías agregar un id a los elementos

andress20 replied:
Listo prop `key` agregado, y id para los elementos con uuid().

simonhoyos commented on alamesa/src/pages/Restaurants.js:
```diff
@@ -0,0 +1,49 @@
+class Restaurants extends Component {
+	
+	state = {
+		name: data
```
Trata de ponerle un nombre más significativo al estado. En vez de name podrías usar restaurants

simonhoyos commented on alamesa/src/Data.js:
```diff
@@ -0,0 +1,19 @@
+export const data = [
+	{
+		name: 'Mister Wok',
```
Sería bueno agregarle más datos a los restaurantes. Podrías agregarle la imagen o el logo del restaurante acá. No importa que todos sean lo mismo, tenerlo dentro del objeto se acerca más a la realidad que tener una imagen quemada en el jsx para todos los elementos

carosalazar28 commented on alamesa/src/pages/Restaurants.js:
```diff
@@ -0,0 +1,49 @@
+	display: grid;
+	grid-template-columns: repeat(4,1fr);
+	grid-grap: 20px;
+`
```
Siempre toca poner **;** al terminar de declarar las reglas de css

andress20 replied:
Listo bien. Ya los agregué.

carosalazar28 commented on alamesa/src/components/Restaurant.js:
```diff
@@ -0,0 +1,49 @@
+const Img = styled.img`
+	float: left;
+	width:  120px;
+  height: 120px;
```
Acá se te fue mal una identación

andress20 replied:
Listo, creo que es cosa de configuración en mi editor. Ya le movi algunas cosas en esa configuración para corregir eso.

andress20 committed:
indentation fixes
add key props on the list
property type button move inside the component jsx button
exchange Object-fit by background-size on Img styled-component
Restaurant logo move to Data.js in order to manage it like props
insert semicolon in every styled-component

andress20 committed:
fix indentation

simonhoyos commented on alamesa/src/components/Restaurant.js:
```diff
@@ -1,45 +1,46 @@
 			<Img 
-				src={RestaurantLogo}
-				alt='Restaurant Logo'
+				src={logo}
```
la identación en este archivo sigue a 4 espacios. Creo que te toca hacer el arreglo manualmente

andress20 replied:
Listo, hice 2 push muy seguidos como 5 minutos de diferencia. Ya le veo bien la identacion simon. de hecho aparece la palabra "Outdated" al frente de la ruta de este archivo en tu comentrio supongo que porque el archivo ya cambio no se. Pero ya puedes volver a revisar.

simonhoyos commented on alamesa/src/pages/Restaurants.js:
```diff
@@ -1,41 +1,44 @@
-		const { name } = this.state
+		const { restaurants } = this.state
 		return (
 			<Container>
```
la identación en este archivo sigue a 4 espacios. Creo que te toca hacer el arreglo manualmente

andress20 replied:
Archivo actualizado!
